pulseguardian/guardian.py

```diff
@@ -219,22 +230,66 @@ def back_to_normal_email(self, user, queue_data):
            queue_data['messages'], self.del_queue_size)
 
         if self.emails and user.email is not None:
-            sendemail(subject=subject, from_addr=config.email_from,
-                      to_addrs=[user.email], username=config.email_account,
-                      password=config.email_password, text_data=body,
-                      server=config.email_smtp_server,
-                      port=config.email_smtp_port,
-                      use_ssl=config.email_ssl)
+            self._sendemail(
+                subject=subject, to_addrs=[user.email], text_data=body)
+
+    def notify_connection_error(self):
+        if self.emails and not self._connection_error_notified:
+            ex_details = traceback.format_exc()
```

Comment:
Hm this is nice because we don't have to pass the exception into the function, but on the other hand it is now an implicit requirement that this function be called during an exception. At least a function docstring explaining this is necessary.